drivers/watchdog/wdt_litex.c

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright (C) 2024 Vogl Electronic GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT litex_watchdog
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/watchdog.h>
+#include <zephyr/device.h>
+#include <zephyr/sys_clock.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(wdt_litex, CONFIG_WDT_LOG_LEVEL);
+
+#include <soc.h>
+
+struct wdt_litex_data {
+	wdt_callback_t callback;
+	uint32_t timeout;
+	bool reset_soc_mode;
+	bool pause_halted;
+};
+
+struct wdt_litex_config {
+	uint32_t control_addr;
+	uint32_t cycles_addr;
+	uint32_t cycles_size;
+	uint32_t remaining_addr;
+	uint32_t ev_status_addr;
+	uint32_t ev_pending_addr;
+	uint32_t ev_enable_addr;
+	void (*irq_cfg_func)(void);
+};
+
+#define CONTROL_FEED_BIT         BIT(0)
+#define CONTROL_ENABLE_BIT       BIT(8)
+#define CONTROL_RESET_BIT        BIT(16)
+#define CONTROL_PAUSE_HALTED_BIT BIT(24)
+
+static bool wdt_litex_is_enabled(const struct device *dev)
+{
+	const struct wdt_litex_config *config = dev->config;
+
+	return litex_read8(config->control_addr) & BIT(0);
+}
+
+static void wdt_litex_irq_enable(const struct device *dev)
+{
+	const struct wdt_litex_config *config = dev->config;
+	struct wdt_litex_data *data = dev->data;
+
+	if (!data->callback) {
+		return;
+	}
+
+	litex_write8(BIT(0), config->ev_pending_addr);
+
+	litex_write8(BIT(0), config->ev_enable_addr);
+}
+
+static void wdt_litex_enable(const struct device *dev)
+{
+	const struct wdt_litex_config *config = dev->config;
+	struct wdt_litex_data *data = dev->data;
+	uint32_t control;
+
+	if (config->cycles_size <= 4) {
+		litex_write32(k_ms_to_cyc_floor32(data->timeout), config->cycles_addr);
+	} else {
+		litex_write64(k_ms_to_cyc_floor64(data->timeout), config->cycles_addr);
+	}
```

Comment:
was this tested? You seem to be using bits as size in the corresponding litex PR: https://github.com/enjoy-digital/litex/pull/1996/files#diff-91072e8d1728a622115f31f10d4a9a6902f27f22054a6236b4b09e00aa97287aR23

Or is it converted to bytes in litex somewhere else

Reply:
It is the size of the reg in the dts, which is in bytes

Reply:
oh, right